### PR TITLE
feat(console): image uploader

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): image uploader

image

## Testing

image
image

Usage:

image

## Checklist

- [ ] `.changset-staged`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/schemas/src/consts/storage.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const maxUploadFileSize = 8 * 1024 * 1024; // 8MB
+
+// Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+export const allowUploadMimeTypes = [
```
Comment (simeng-li) [inline packages/schemas/src/consts/storage.ts:4]: Should we put it under the schemas? It seems unrelated to the main auth flow. More like silverhand essentials or sth

File: packages/console/src/components/FileUploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    disabled: isUploading,
+  });
+
+  return (
+    <div>
+      <div
+        {...getRootProps()}
```
Comment (simeng-li) [inline packages/console/src/components/FileUploader/index.tsx:92]: nit: Does it handle the on-click event?

File: packages/console/src/components/FileUploader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  const [uploaderError, setUploaderError] = useState<string>();
+  const hasError = Boolean(uploaderError);
+
+  const api = useApi();
+
+  const onDrop = useCallback(
+    async (acceptedFiles: File[]) => {
+      setUploaderError(undefined);
+
+      if (acceptedFiles.length > allowedFileCount) {
```
Comment (simeng-li) [inline packages/console/src/components/FileUploader/index.tsx:35]: Should we replace the source directly?

File: packages/console/src/components/FileUploader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        const supportedFileTypes = allowedMimeTypes.map((type) =>
+          type.split('/')[1]?.toUpperCase()
+        );
+        setUploaderError(t('components.uploader.error_file_type', { types: supportedFileTypes }));
+
+        return;
+      }
+
+      const maxSizeInBytes = maxSize * 1024;
+      const fileSizeLimit = maxSizeInBytes < maxUploadFileSize ? maxSizeInBytes : maxUploadFileSize;
```
Comment (simeng-li) [inline packages/console/src/components/FileUploader/index.tsx:57]: Math.min

File: packages/console/src/components/FileUploader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      if (!allowedMimeTypes.map(String).includes(selectedFile.type)) {
+        const supportedFileTypes = allowedMimeTypes.map((type) =>
+          type.split('/')[1]?.toUpperCase()
+        );
+        setUploaderError(t('components.uploader.error_file_type', { types: supportedFileTypes }));
+
+        return;
+      }
+
+      const maxSizeInBytes = maxSize * 1024;
```
Comment (simeng-li) [inline packages/console/src/components/FileUploader/index.tsx:56]: Do we have a reference for the unit we use for the file size related props?  e.g. better align the unit of the input props and `maxUploadFileSize`.

File: packages/console/src/components/FileUploader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      const fileSizeLimit = maxSizeInBytes < maxUploadFileSize ? maxSizeInBytes : maxUploadFileSize;
+
+      if (selectedFile.size > fileSizeLimit) {
+        setUploaderError(t('components.uploader.error_file_size', { count: fileSizeLimit / 1024 }));
```
Comment (simeng-li) [inline packages/console/src/components/FileUploader/index.tsx:60]: ditto:  first * 1024, then / 1024 looks redundant.

File: packages/schemas/src/consts/storage.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/schemas/src/consts/storage.ts:4]: It's ok to put it under the schemas since this array is also used to construct types(DTO) in both the backend and front end.

File: packages/console/src/components/FileUploader/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/components/FileUploader/index.tsx:91]: Yes.

File: packages/console/src/components/FileUploader/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/components/FileUploader/index.tsx:35]: It may be confusing when the user drops more than two files.
We will have errors to let the user know what is happening.

File: packages/console/src/components/FileUploader/index.tsx (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/components/FileUploader/index.tsx]: Updated.

File: packages/console/src/components/FileUploader/index.tsx (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/components/FileUploader/index.tsx]: Updated the maxSize unit from `KB` to `Bytes`, and the `maxSize * 1024` can be removed.

File: packages/console/src/components/FileUploader/index.tsx (hunk 6 above)
Comment (xiaoyijun) [reply packages/console/src/components/FileUploader/index.tsx:59]: updated.

File: packages/core/src/routes/user-assets.ts [hunk 7]
```diff
@@ -21,8 +19,7 @@ import type { AuthedRouter, RouterInitArgs } from './types.js';
 export default function userAssetsRoutes<T extends AuthedRouter>(...[router]: RouterInitArgs<T>) {
   router.get('/user-assets/service-status', async (ctx, next) => {
     const { storageProviderConfig } = SystemContext.shared;
-
-    ctx.body = storageProviderConfig
+    const status: UserAssetsServiceStatusResponse = storageProviderConfig
```
Comment (wangsijie) [inline packages/core/src/routes/user-assets.ts:22]: would you mind add this to response guard, thanks

**Review by wangsijie — APPROVED** [review]
